### PR TITLE
Visium wrangle sample ntbk

### PR DESCRIPTION
**Pull request by jahilton**
establish a notebook that contributors and curators can use to curate Visium Datasets to CELLxGENE standards
Notably does not include is_primary_data & gene mapping/filtering as we typically deal with those things rather than contributors

### CODE REVIEW

File: cellxgene_resources/curation_visium.ipynb [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+   "source": [
+    "Specify the folder that includes these files...\n",
+    "- raw_feature_bc_matrix.h5\n",
+    "- spatial/\n",
+    "  - scalefactors_json.json\n",
+    "  - tissue_hires_image.png\n",
+    "  - tissue_lowres_image.png*\n",
+    "  - tissue_positions_list.csv\n",
+    "\n",
+    "*the lowres image will be deleted but is required for `read_visium()`"
```
Comment (brian-mott) [inline cellxgene_resources/curation_visium.ipynb:29]: I think having this url https://www.ebi.ac.uk/biostudies/arrayexpress/studies/E-MTAB-12698 and some further comments that the output files and full res tiff are here would be helpful so someone could run the notebook on these files and see the output.

File: cellxgene_resources/curation_visium.ipynb [hunk 2]
```diff
@@ -0,0 +1,227 @@
...
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "folder = 'WS_PLA_S9101764'\n",
+    "\n",
+    "#REQUIRED to include empty spots, so specify raw_feature_bc_matrix.h5\n",
+    "adata = sc.read_visium(folder, count_file='raw_feature_bc_matrix.h5')"
```
Comment (brian-mott) [inline cellxgene_resources/curation_visium.ipynb:41]: The code block below to fill in empty 0s initially failed until I ran `adata.var_names_make_unique()`. I think further comment to clarify to run this method if the size is < 4992 and there is a warning that variable names are not unique.

File: cellxgene_resources/curation_visium.ipynb [hunk 3]
```diff
@@ -0,0 +1,227 @@
...
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#map in cell_types\n",
+    "#this example reads in a csv with barcodes in the first column and CL terms in the second column\n",
```
Comment (brian-mott) [inline cellxgene_resources/curation_visium.ipynb:154]: And a little more to clarify that a csv in this format can be provided to add in the cell ontology but it is not part of the output files or found in this repo

**Review by brian-mott — CHANGES_REQUESTED** [review]
I was able to run this notebook on a few outputs from the url in this review and on another 10X mouse dataset that had less than 4992 observations. I think it would be helpful to clarify the comments in these few places and then we can see what other contributors think as they make use of this notebook

File: cellxgene_resources/curation_visium.ipynb (hunk 1 above)
Comment (jahilton) [reply cellxgene_resources/curation_visium.ipynb]: Addressed, mostly in a new notebook header

File: cellxgene_resources/curation_visium.ipynb (hunk 2 above)
Comment (jahilton) [reply cellxgene_resources/curation_visium.ipynb]: Good catch. Fixed by setting gene_ids as the index to start.

File: cellxgene_resources/curation_visium.ipynb (hunk 3 above)
Comment (jahilton) [reply cellxgene_resources/curation_visium.ipynb:165]: Added additional comments on this. Let me know what you think

File: cellxgene_resources/curation_visium.ipynb (hunk 2 above)
Comment (brian-mott) [reply cellxgene_resources/curation_visium.ipynb]: Looks good, looks like this now works correctly when obs is < 4992

File: cellxgene_resources/curation_visium.ipynb (hunk 3 above)
Comment (brian-mott) [reply cellxgene_resources/curation_visium.ipynb:165]: This looks good too, hopefully will be clear for contributors.

**Review by brian-mott — APPROVED** [review]
LGTM, further clear comments and setting index resolves previous error to add 0s

**Pull request merged**